tests/core/test_callable_utils.py

```diff
@@ -185,6 +190,12 @@ def func_with_extra_param_in_docstring(a: int) -> str:
     """
 
     assert sql_body.strip() == expected_sql.strip()
+    assert len(recwarn) == 1
+    warn = recwarn.pop(UserWarning)
+    assert (
+        "The following parameters are documented in the docstring but not present in the function signature: b"
+        in str(warn.message)
+    )
```

Comment:
we can use `pytest.warns` instead.

Reply:
I shifted over to pytest.warns since it's the standard we use in MLflow. The only reason why I don't like using it for these cases is that it's WAY more verbose for zero gain. 
For sake of consistency and familiarity, migrated to pytest.